HIP/hip-0000-event-serialization.md

```diff
@@ -0,0 +1,227 @@
+| Hash Object                                                            |
++------------+---------+---------------------------------+---------------+
+| # of bytes | Type    | Description                     | Typical Value |
++------------+---------+---------------------------------+---------------+
+| 4 bytes    | Integer | Serializaiton version of object | 0x00000001    |
++------------+---------+---------------------------------+---------------+
+| 4 bytes    | Integer | Hash type                       | 0x58ff811b    |
++------------+---------+---------------------------------+---------------+
+| 4 bytes    | Integer | Length of hash                  | 0x00000030    |
++------------+---------+---------------------------------+---------------+
+| N bytes    | byte[]  | Hash value (N = length of hash) |               |
++------------+---------+---------------------------------+---------------+
+```
+
+### EventDescriptor
+
+```
++------------------------------------------------------------------------+
+| EventDescriptor Object                                                 |
++------------+---------+---------------------------------+---------------+
+| # of bytes | Type    | Description                     | Typical Value |
++------------+---------+---------------------------------+---------------+
+| 4 bytes    | Integer | Serialization version of object | 0x00000003    |
++------------+---------+---------------------------------+---------------+
+| H bytes    | Hash    | The hash of the event           |               |
++------------+---------+---------------------------------+---------------+
+| 12 bytes   | NodeId  | The node that created the event |               |
++------------+---------+---------------------------------+---------------+
+| 8 bytes    | Long    | The generation of the event     |               |
++------------+---------+---------------------------------+---------------+
+| 8 bytes    | Long    | The roster round of the event   |               |
++------------+---------+---------------------------------+---------------+
+```
+
+### BaseEventHashedData
+
+```
++--------------------------------------------------------------------------------------------------+
+| BaseEventHashedData Object                                                                       |
++------------+-----------------------+---------------------------------------------+---------------+
+| # of bytes | Type                  | Description                                 | Typical Value |
++------------+-----------------------+---------------------------------------------+---------------+
+| 4 bytes    | Integer               | Serialization version of object             | 0x00000004    |
++------------+-----------------------+---------------------------------------------+---------------+
+| V bytes    | SoftwareVersion       | The software version that created the event |               |
```

Comment:
Why isn't `SoftwareVersion` a predetermined number of bytes?

Reply:
In testing the software version is an integer.   In deployment, the software version is a variable length string.

Reply:
Nevermind, I see you now say it's a string in the table